### PR TITLE
Organization ownership

### PR DESCRIPTION
**Pull request by thesuzerain**
Organiztion owner structure changed as discussed.
- is_owner is unique
- projects owned by an org do not have a owner
- projects owned by an org do not have that orgs owner as a member (so permissions cant be overwritten)
- removing a project from an org you must specify to whom ownership of it goes
- when deleting an org, all projects become owned by the org owner.

Review carefully as it deals with ownership /permissions.

### CODE REVIEW

File: src/routes/v3/organizations.rs [hunk 1]
```diff
@@ -522,7 +523,105 @@ pub async fn organization_delete(
...
+        SET is_owner = FALSE
+        WHERE tm.team_id = ANY($1)
+        ",
+        &organization_project_teams
+            .iter()
+            .map(|x| x.0 as i64)
+            .collect::<Vec<_>>()
+    )
+    .execute(&mut *transaction)
+    .await?;
```
Comment (Geometrically) [inline src/routes/v3/organizations.rs:606]: this should be removed since it's an illegal state

File: src/routes/v3/organizations.rs [hunk 2]
```diff
@@ -522,7 +523,105 @@ pub async fn organization_delete(
...
+
+    for project_team_without_owner in project_teams_without_owner {
+        let new_id =
+            crate::database::models::ids::generate_team_member_id(&mut transaction).await?;
+        let member = TeamMember {
+            id: new_id,
+            team_id: project_team_without_owner,
+            user_id: owner_id,
+            role: "Inherited Owner".to_string(),
+            is_owner: false,
```
Comment (Geometrically) [inline src/routes/v3/organizations.rs:583]: set is_owner here

Comment (Geometrically) [reply src/routes/v3/organizations.rs:583]: to true

File: src/routes/v3/organizations.rs [hunk 3]
```diff
@@ -522,7 +523,105 @@ pub async fn organization_delete(
...
+        ",
+        &organization_project_teams
+            .iter()
+            .map(|x| x.0 as i64)
+            .collect::<Vec<_>>(),
+        owner_id.0
+    )
+    .execute(&mut *transaction)
+    .await?;
+
```
Comment (Geometrically) [inline src/routes/v3/organizations.rs:623]: remove this (also make it so the owner of an org can never be a team member of the project

File: src/routes/v3/organizations.rs [hunk 4]
```diff
@@ -622,6 +725,23 @@ pub async fn organization_projects_add(
...
+                "
+                UPDATE team_members
+                SET is_owner = FALSE
+                WHERE id = $1
+                ",
+                team_member_id as database::models::ids::TeamMemberId
+            )
+            .execute(&mut *transaction)
+            .await?;
+        }
```
Comment (Geometrically) [inline src/routes/v3/organizations.rs:743]: also shouldn't be needed

File: src/routes/v3/organizations.rs [hunk 5]
```diff
@@ -696,7 +824,78 @@ pub async fn organization_projects_remove(
...
+            organization.id,
+            data.new_owner.into(),
+            &**pool,
+        )
+        .await?
+        .ok_or_else(|| {
+            ApiError::InvalidInput(
+                "The specified user is not a member of this organization!".to_string(),
+            )
+        })?;
```
Comment (Geometrically) [inline src/routes/v3/organizations.rs:838]: the user should be allowed to move it to a user who is part of the project's team but not the org's team as well

File: src/routes/v3/organizations.rs [hunk 6]
```diff
@@ -696,7 +824,78 @@ pub async fn organization_projects_remove(
...
+        sqlx::query!(
+            "
+            UPDATE team_members
+            SET is_owner = TRUE
+            WHERE (id = $1)
+            ",
+            new_owner.id as database::models::ids::TeamMemberId
+        )
+        .execute(&mut *transaction)
+        .await?;
```
Comment (Geometrically) [inline src/routes/v3/organizations.rs:897]: also see above, since it's redundant we can probs get rid of it

File: src/routes/v3/teams.rs [hunk 7]
```diff
@@ -723,8 +724,14 @@ pub async fn transfer_ownership(
     .await?
     .1;
 
+    println!(
+        "Transferring owner from {} to {}",
+        current_user.id.0, new_owner.user_id.0
+    );
+    println!("Trasnfering team: {}", id.0);
```
Comment (Geometrically) [inline src/routes/v3/teams.rs:731]: print statements

**Review by Geometrically — CHANGES_REQUESTED** [review]

File: src/routes/v3/organizations.rs (hunk 3 above)
Comment (thesuzerain) [reply src/routes/v3/organizations.rs:623]: and also there shouldnt be an owner of a project if there is an organization

**Review by Geometrically — APPROVED** [review]

**Pull request merged**